src/server/routes/oauth.ts

```diff
@@ -256,38 +181,51 @@ const authenticationHandler = async (
 			);
 		} else if (refreshToken && !legacy_identity_id) {
 			try {
-				// We have a refresh token and no legacy IDAPI ID, so we can try to fix this user.
-				const validTokenSet = await fetchValidTokenSet({
+				// We have a refresh token and no legacy IDAPI ID, which means the user is
+				// in one of two states:
+				// 1. They exist in Okta _AND_ IDAPI, but the two accounts aren't linked (i.e.
+				//    the legacyIdentityId field is missing from the Okta user profile).
+				// 2. They existed in Okta but _NOT_ in IDAPI - which will have been fixed
+				//    in the exchangeAccessTokenForCookies call above.
+
+				// Run the fix function to link the two accounts. This won't be needed
+				// in case (2) above, but it's cheaper to just run it than to check first.
```

Comment:
as far as I know we are not sure why sometimes users don't have idapi ids in okta even though they have idapi accounts. Given that the function the okta hook calls is pretty much the same as we call manually here I'm not 100% sure the user has the idapi id in okta at this point.

Reply:
I was thinking maybe the idapi call returns success in some cases even though the okta function didn't actually update the user profile ?